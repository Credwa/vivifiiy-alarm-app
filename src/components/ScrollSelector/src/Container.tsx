import React, { FC } from 'react';
import { View } from 'react-native';

interface IProps {
  wrapperHeight: number;
  wrapperWidth: number | string;
  wrapperBackground: string;
}

const Container: FC<IProps> = ({ children, wrapperBackground, wrapperWidth, wrapperHeight }) => (
  <View
    style={{
      height: wrapperHeight,
      overflow: 'hidden',
      alignSelf: 'center',
      width: wrapperWidth,
      backgroundColor: wrapperBackground
    }}
  >
    {children}
  </View>
);

export default Container;
