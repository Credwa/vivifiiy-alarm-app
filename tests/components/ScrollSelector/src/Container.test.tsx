import React from 'react';
import renderer from 'react-test-renderer';
import Container from '@/components/ScrollSelector/src/Container';

describe('ScrollSelector/src/ <Container />', () => {
  it(`renders correctly`, () => {
    const tree = renderer
      .create(<Container wrapperHeight={180} wrapperWidth={150} wrapperBackground={'red'} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
