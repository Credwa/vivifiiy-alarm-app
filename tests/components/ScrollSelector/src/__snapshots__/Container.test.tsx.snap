// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollSelector/src/ <Container /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "red",
      "height": 180,
      "overflow": "hidden",
      "width": 150,
    }
  }
/>
`;
